### PR TITLE
Add ability to handle RNA sequence data types

### PR DESCRIPTION
**Pull request by mikerobeson**
This PR adds types, formats, and transformers to handle RNA sequence data. The additions / changes are:

- Changed `DNAFASTAFormat` 
   - was updated so that it *only* accepts DNA sequences, i.e. disallows `U`
   - This is a problem as it is possible to import RNA sequence data upon import, be incorrectly called a DNA Sequence type, only to fail downstream using other QIIME 2 operations.
- Added `RNAFASTAFormat` 
   - only accepts RNA sequences, disallows `T` (applies to all RNA formats)
- Added `AlignedRNAFASTAFormat`
- Added `RNASequencesDirectoryFormat`
- Added `AlignedRNASequencesDirectoryFormat` 
- Added `PairedRNASequencesDirectoryFormat`
- Added Type `FeatureData[RNASequence]`
- Added Type `FeatureData[PairedEndRNASequence]`
- Added Type `FeatureData[AlignedRNASequence]`
- Added a `NucleicAcidIterator` class to transformers. 
   - From which all other `DNA*` and `RNA*` iterators Inherit.

### CODE REVIEW

File: q2_types/feature_data/_format.py [hunk 1]
```diff
@@ -271,20 +271,37 @@ def _validate_line_lengths(
 class DNAFASTAFormat(FASTAFormat):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.alphabet = "ACGTURYKMSWBDHVN"
+        self.alphabet = "ACGTRYKMSWBDHVN"
```
Comment (misialq) [inline q2_types/feature_data/_format.py:274]: 🙏

File: q2_types/feature_data/tests/test_format.py [hunk 2]
```diff
@@ -297,6 +299,131 @@ def test_aligned_dna_sequences_directory_format(self):
...
+        filepath = os.path.join(self.temp_dir.name, 'empty')
+        with open(filepath, 'w') as fh:
+            fh.write('\n')
+        format = RNAFASTAFormat(filepath, mode='r')
+
+        format.validate()
+
+    def test_rna_fasta_format_invalid_characters(self):
+        filepath = self.get_data_path('not-rna-sequences.fasta')
+        format = DNAFASTAFormat(filepath, mode='r')
```
Comment (misialq) [inline q2_types/feature_data/tests/test_format.py:325]:
```suggestion
        format = RNAFASTAFormat(filepath, mode='r')
```

**Review by misialq — CHANGES_REQUESTED** [review]
One small thing aside, LGTM!

File: q2_types/feature_data/tests/test_format.py (hunk 2 above)
Comment (mikerobeson) [reply q2_types/feature_data/tests/test_format.py:325]: Ohh good catch!

**Review by misialq — APPROVED** [review]
🚀

**Pull request merged**